src/code/padmgr.c

```diff
@@ -94,164 +172,191 @@ void PadMgr_RumbleControl(PadMgr* padMgr) {
+ */
+void PadMgr_RumbleSet(PadMgr* padMgr, u8 enable[MAXCONTROLLERS]) {
     s32 i;
 
-    for (i = 0; i < 4; i++) {
-        padMgr->rumbleEnable[i] = ctrlrRumbles[i];
+    for (i = 0; i < MAXCONTROLLERS; i++) {
+        padMgr->rumbleEnable[i] = enable[i];
     }
 
-    padMgr->rumbleOnFrames = 240;
+    padMgr->rumbleOnTimer = 240;
 }
 
-void PadMgr_ProcessInputs(PadMgr* padMgr) {
+/**
+ * Updates `padMgr->inputs` based on the error response of each controller
+ */
+void PadMgr_UpdateInputs(PadMgr* padMgr) {
     s32 i;
     Input* input;
-    OSContPad* padnow1; // original name
+    OSContPad* pad; // original name: "padnow1"
     s32 buttonDiff;
 
     PadMgr_LockPadData(padMgr);
 
-    input = &padMgr->inputs[0];
-    padnow1 = &padMgr->pads[0];
-
-    for (i = 0; i < padMgr->nControllers; i++, input++, padnow1++) {
+    for (input = &padMgr->inputs[0], pad = &padMgr->pads[0], i = 0; i < padMgr->nControllers; i++, input++, pad++) {
         input->prev = input->cur;
 
         if (1) {} // Necessary to match
 
-        switch (padnow1->errno) {
+        switch (pad->errno) {
             case 0:
-                input->cur = *padnow1;
+                // No error, copy inputs
+                input->cur = *pad;
                 if (!padMgr->ctrlrIsConnected[i]) {
                     padMgr->ctrlrIsConnected[i] = true;
                     osSyncPrintf(VT_FGCOL(YELLOW));
                     osSyncPrintf("padmgr: %dコン: %s\n", i + 1, "認識しました"); // "Recognized"
                     osSyncPrintf(VT_RST);
                 }
                 break;
-            case 4:
+            case (CHNL_ERR_OVERRUN >> 4):
+                // Overrun error, reuse previous inputs
                 input->cur = input->prev;
                 LOG_NUM("this->Key_switch[i]", padMgr->ctrlrIsConnected[i], "../padmgr.c", 380);
                 osSyncPrintf(VT_FGCOL(YELLOW));
                 // "Overrun error occurred"
                 osSyncPrintf("padmgr: %dコン: %s\n", i + 1, "オーバーランエラーが発生");
                 osSyncPrintf(VT_RST);
                 break;
-            case 8:
+            case (CHNL_ERR_NORESP >> 4):
+                // No response error, take inputs as 0
                 input->cur.button = 0;
                 input->cur.stick_x = 0;
                 input->cur.stick_y = 0;
-                input->cur.errno = padnow1->errno;
+                input->cur.errno = pad->errno;
                 if (padMgr->ctrlrIsConnected[i]) {
+                    // If we get no response, consider the controller disconnected
                     padMgr->ctrlrIsConnected[i] = false;
-                    padMgr->pakType[i] = 0;
-                    padMgr->rumbleCounter[i] = 0xFF;
+                    padMgr->pakType[i] = CONT_PAK_NONE;
+                    padMgr->rumbleTimer[i] = 0xFF;
                     osSyncPrintf(VT_FGCOL(YELLOW));
                     // "Do not respond"?
                     osSyncPrintf("padmgr: %dコン: %s\n", i + 1, "応答しません");
```

Comment:
Looks more like "not responding", there's some examples of usage of this phrase here: https://context.reverso.net/%E7%BF%BB%E8%A8%B3/%E6%97%A5%E6%9C%AC%E8%AA%9E-%E8%8B%B1%E8%AA%9E/%E5%BF%9C%E7%AD%94%E3%81%97%E3%81%BE%E3%81%9B%E3%82%93